src/model.js

```diff
@@ -134,10 +139,15 @@ const BookshelfModel = ModelBase.extend({
    *   be the singular form of this model's tableName, followed by `_id` /
    *   `_{{{@link Model#idAttribute idAttribute}}}`.
    *
+   * @param {string=} foreignKeyTarget
+   *
+   *   Column in the `Target` model's table which `foreignKey` references, if other
+   *   than `Target` model's `id` / `{@link Model#idAttribute idAttribute}`.
+   *
    * @returns {Collection}
    */
-  hasMany(Target, foreignKey) {
-    return this._relation('hasMany', Target, {foreignKey}).init(this);
+  hasMany(Target, foreignKey, foreignKeyTarget) {
+    return this._relation('hasMany', Target, {foreignKey, foreignKeyTarget}).init(this);
```

Comment:
In my [comment](https://github.com/tgriesser/bookshelf/pull/1397#issuecomment-250721474) I meant changing this to

```
{foreignKey, targetIdAttribute: foreignKeyTarget}
```